content/docs/tasks/traffic-management/tcp-traffic-shifting/index.md

```diff
@@ -0,0 +1,172 @@
+---
+title: TCP Traffic Shifting
+description: Shows you how to migrate TCP traffic from an old to new version of a TCP service.
+weight: 25
+keywords: [traffic-management,tcp-traffic-shifting]
+aliases:
+    - /docs/tasks/traffic-management/tcp-version-migration.html
+---
+
+This task shows you how to gradually migrate TCP traffic from one version of a
+microservice to another. For example, you might migrate TCP traffic from an
+older version to a new version.
+
+A common use case is to migrate TCP traffic gradually from one version of a
+microservice to another. In Istio, you accomplish this goal by configuring a
+sequence of rules that route a percentage of TCP traffic to one service or
+another. In this task, you will send 100% of the TCP traffic to `tcp-echo:v1`.
+Then, you will route 20% of the TCP traffic to `tcp-echo:v2` using Istio's
+weighted routing feature.
+
+## Before you begin
+
+* Setup Istio by following the instructions in the [Installation guide](/docs/setup/).
+
+* Review the [Traffic Management](/docs/concepts/traffic-management) concepts doc.
+
+## Apply weight-based TCP routing
+
+1.  To get started, deploy the `v1` version of the `tcp-echo` microservice.
+
+    *   If you are using [manual sidecar injection](/docs/setup/kubernetes/sidecar-injection/#manual-sidecar-injection),
+        use the following command
+
+        {{< text bash >}}
+        $ kubectl apply -f <(istioctl kube-inject -f @samples/tcp-echo/tcp-echo-services.yaml@)
+        {{< /text >}}
+
+        The `istioctl kube-inject` command is used to manually modify the `tcp-echo-services.yaml`
+        file before creating the deployments as documented [here](/docs/reference/commands/istioctl/#istioctl-kube-inject).
+
+    *   If you are using a cluster with
+        [automatic sidecar injection](/docs/setup/kubernetes/sidecar-injection/#automatic-sidecar-injection)
+        enabled, label the `default` namespace with `istio-injection=enabled`
+
+        {{< text bash >}}
+        $ kubectl label namespace default istio-injection=enabled
+        {{< /text >}}
+
+        Then simply deploy the services using `kubectl`
+
+        {{< text bash >}}
+        $ kubectl apply -f @samples/tcp-echo/tcp-echo-services.yaml@
+        {{< /text >}}
+
+1.  Next, route all TCP traffic to the `v1` version of the `tcp-echo` microservice.
+
+    {{< text bash >}}
+    $ kubectl apply -f @samples/tcp-echo/tcp-echo-all-v1.yaml@
+    {{< /text >}}
+
+1.  Confirm that the `tcp-echo` service is up and running.
+
+    The `$INGRESS_HOST` variable below is the External IP address of the ingress, as explained in
+the [Bookinfo](/docs/examples/bookinfo/#determining-the-ingress-ip-and-port) doc. To obtain the
+`$INGRESS_PORT` value, use the following command.
```

Comment:
Why not just tell user to set the INGRESS_HOST and INGRESS_PORT variables according to [these instructions](https://istio.io/docs/tasks/traffic-management/ingress/#determining-the-ingress-ip-and-ports)?

Reply:
The document you mentioned only speaks about `http2` and `https`, while we're working with the `tcp` port here.